Convert type comments to annotations in caffe2/test/onnx/

stroxler opened the pull request:
## Summary

Convert type comments in caffe2/test/onnx/

Produced by running:
```
python -m  libcst.tool codemod convert_type_comments.ConvertTypeComment caffe2/test/onnx/
```
from the parent directory.

One question is whether we actually want to scrap type comment here. There are some jit tests where we're explicitly aiming to validate py2-style type comments; I don't think this test is one of those cases but if I'm misreading it I can close the PR.


release notes:

Converts type comments to annotations.

topics:

Refactor

stroxler commented:
Hoping to get a look from @BowenBao or @garymm to make sure converting type commects to annotations is okay

garymm commented:
@stroxler can we avoid unrelated reformatting in this PR?
In the past I used [com2ann](https://pypi.org/project/com2ann/) and it can do this without reformatting everything so drastically.

I can't even spot the type annotation related diffs because of all of the formatting diffs.
If you want to reformat, can you please do that separately?

stroxler commented:
Sure, I can back out the formatting.

Our codemod tools pick up that all Meta code is supposed to be black-formatted which is generally useful because the transform can otherwise introduce long lines, but I can see we don't like black here.

It might still be worth running black, which I could do in a separate PR. Otherwise sooner or later other codemods will likely hit the same issue; our internal python tools are getting more standardized and opinionated as time goes on.

stroxler committed:
Convert type comments to annotations in caffe2/test/onnx/ (#72632)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/72632

Convert type comments in caffe2/test/onnx/

Produced by running:
```
python -m  libcst.tool codemod convert_type_comments.ConvertTypeComment caffe2/test/onnx/
```
from fbsource.

See
https://fb.workplace.com/groups/pythonfoundation/permalink/3106231549690303/

Test Plan:
This should be safe to land - the only potential for actual breakage
is if the code is using runtime annotations (which is very unlikely
given that we're only modifying modules that use type comments!)

waitforsandcastle

Reviewed By: msaroufim

Differential Revision: D34112196

fbshipit-source-id: 595dacf3881588451465bb218f8b11816812a959

BowenBao commented on test/onnx/test_pytorch_onnx_onnxruntime.py:
```diff
@@ -7654,8 +7652,11 @@ def forward(self, input1, input2):
+                self,
+                input1: Tuple[Tensor, Tensor],
+                input2: Tuple[Tensor, Tensor],
+            ) -> Tuple[Tensor, Tensor]:
```
failed in ci
```suggestion
            ) -> Tuple[Tuple[Tensor, Tensor], Tuple[Tensor, Tensor]]:
```

stroxler committed:
Update test/onnx/test_pytorch_onnx_onnxruntime.py

Co-authored-by: Bowen Bao <imbowenbao@outlook.com>

garymm commented:
@stroxler I'd love it if the entire PyTorch code base was formatted with black, happy to approve changes that are part of that.

stroxler commented:
@garymm Sweet - I'll ask around about that and see if it's something the maintainers would be onboard with; there may be some pushback because it can hurt readability of code that isn't written with formatting in mind.
